Explorer/Assets/DCL/Input/UnityInputSystem/Blocks/InputBlock.cs

```diff
@@ -0,0 +1,59 @@
+using Arch.Core;
+using DCL.CharacterMotion.Components;
+using DCL.Utilities;
+using System;
+
+namespace DCL.Input.UnityInputSystem.Blocks
+{
+    public class InputBlock : IInputBlock
+    {
+        private readonly ObjectProxy<DCLInput> dclInput;
+        private readonly ObjectProxy<World> globalWorld;
+        private readonly ObjectProxy<Entity> playerEntity;
+
+        public InputBlock(ObjectProxy<DCLInput> dclInput, ObjectProxy<World> globalWorld, ObjectProxy<Entity> playerEntity)
+        {
+            this.dclInput = dclInput;
+            this.globalWorld = globalWorld;
+            this.playerEntity = playerEntity;
+        }
+
+        public void BlockMovement()
+        {
+            World().AddOrGet(PlayerEntity(), new MovementBlockerComponent());
+            DclInput().Shortcuts.Disable();
+            DclInput().Camera.Disable();
+        }
+
+        public void UnblockMovement()
+        {
+            World().Remove<MovementBlockerComponent>(PlayerEntity());
+            DclInput().Shortcuts.Enable();
+            DclInput().Camera.Enable();
+        }
+
+        private World World()
```

Comment:
Please rename methods into verbs, like `GetWorld`, otherwise they should be properties